namespace/namespace.go

```diff
@@ -29,8 +28,8 @@ func PrefixTransform(prefix ds.Key) ktds.KeyTransform {
 				panic("expected prefix not found")
 			}
 
-			s := strings.TrimPrefix(k.String(), prefix.String())
-			return ds.NewKey(s)
+			s := k.String()[len(prefix.String()):]
```

Comment:
I could but then what, panic.  Also it is redundant because if prefix.IsAncestorOf passes then there is no way for this to be out of bounds.